src/atlas/array/native/NativeDataStore.h

```diff
@@ -241,9 +241,49 @@ class DataStore : public ArrayDataStore {
 template <typename Value>
 class WrappedDataStore : public ArrayDataStore {
 public:
+    WrappedDataStore(Value* host_data, const ArraySpec& spec, size_t size): host_data_(host_data), size_(size),
+        device_memory_resource_(memory::device::traced_resource()),
+        device_allocator_{device_memory_resource_.get()}
+    {
+        init_device();
+        contiguous_ = spec.contiguous();
+        if (! contiguous_) {
+            int break_idx = 0;
+            size_t shp_mult_rhs = spec.shape()[spec.rank() - 1];
+            for (int i = spec.rank() - 1; i > 0; i--) {
+                if (shp_mult_rhs != spec.strides()[i - 1]) {
+                    break_idx = i - 1;
+                    break;
+                }
+                shp_mult_rhs *= spec.shape()[i - 1];
+            }
+            size_t shp_mult_lhs = spec.shape()[0];
+            for (int i = 1; i <= break_idx; i++) {
+                shp_mult_lhs *= spec.shape()[i];
+            }
+            if (spec.strides()[spec.rank() - 1] > 1) {
+                memcpy_h2d_pitch_ = 1;
+                memcpy_d2h_pitch_ = spec.strides()[spec.rank() - 1];
+                memcpy_width_ = 1;
+                memcpy_height_ = spec.shape()[0] * spec.device_strides()[0];
+            }
+            else {
+                memcpy_h2d_pitch_ = shp_mult_rhs;
+                memcpy_d2h_pitch_ = spec.strides()[break_idx];
+                memcpy_width_ = shp_mult_rhs;
+                memcpy_height_ = shp_mult_lhs;
+            }
+        }
+    }
+
     WrappedDataStore(Value* host_data, size_t size): host_data_(host_data), size_(size),
```

Comment:
This constructor does not set a default value for `contiguous_`.
I don't think this constructor is used anymore as its use is changed for the one using `spec`, but regardless good to keep and fix.

Reply:
Indeed, thank you. `contiguous_` is now defaulted to `true`, except if `spec` is used in the constructor and then it is `spec.contiguous()`. See 6811aa37f.